File: KICKSTART Google/problemC.cpp
#include<bits/stdc++.h>
#define deb(x) cerr<<#x<<" "<<x<<endl;
#define ll long long
using namespace std;

int main(){
	#ifndef ONLINE_JUDGE
	freopen("input.txt","r",stdin);
	freopen("output.txt","w",stdout);
	#endif
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	ll t;
	cin>>t;
	ll temp=t;
	while(t--){
		string s;
		cin>>s;
		ll n= s.length();
		ll multiplier=1;
		stack<ll> indices;
		unordered_map<char,ll> freq;
		for(ll i=0;i<n;i++){
			if((int)s[i]>48 and (int)s[i]<58){
				indices.push(s[i]-'0');
				multiplier*=(s[i]-'0');
			}
			else if(s[i]==')'){
				ll top= indices.top();
				indices.pop();
				multiplier/=top;
			}
			else if(s[i]=='(')
				continue;
			else{
				freq[s[i]]+=(multiplier);
			}

		}
		
		for (auto m:freq)
		{
			cout<<m.first<<" "<<m.second<<"\n";
		}
		ll x=1+freq['S']-freq['N'];
		ll y=1+freq['E']-freq['W'];
		if(x<=0)
			x+=1000000000;
		else if(x>1000000000){
			x-=1000000000;
		}
		if(y<=0)
			y+=1000000000;
		else if(y>1000000000){
			y-=1000000000;
		}
		

		cout<<"Case #"<<temp-t<<": "<<y<<" "<<x<<"\n";
	}



	 
return 0;
}

File: KICKSTART Google/problemC.py
t=int(input())
temp=t
# t-=1
while(t):
    s=input()
    multiplier=int(1)
    stack=[]
    freq={}
    for m in s:
        if(ord(m)>48 and ord(m)<58):
            multiplier*=int(m)
            # print("mul",multiplier)
            stack.append(int(m))
        elif (m==')'):
            k=stack.pop()
            multiplier=multiplier//k
        elif (m=='('):
            continue
        else:
            if(m in freq.keys()):
                freq[m]+=multiplier
            else:
                freq[m]=multiplier
    if('S' not in freq.keys()):
        freq['S']=0
        # print('s')
    if('N' not in freq.keys()):
        freq['N']=0
        # print('n')

    if('E' not in freq.keys()):
        freq['E']=0
        # print('e')

    if('W' not in freq.keys()):
        freq['W']=0
        # print('w')

    x= freq['S']-freq['N']+1
    y=freq['E']-freq['W']+1
    # x=-1000000000132
    if(x>1000000000):
        x=x%1000000000
    elif(x<=0):
        xpos=abs(x)
        xpos= xpos%1000000000
        x=1000000000-xpos
    if(y>1000000000):
        y=y%1000000000
    elif(y<=0):
        ypos=abs(y)
        ypos= ypos%1000000000
        y=1000000000-ypos



    # for k in freq.keys():
    #     print(k,freq[k])
    ans="Case #"+str(int(temp-t+1))+": "+str(int(y))+" "+str(int(x))
    print(ans)
    t-=1
